ganga/GangaCore/test/GPI/TestAddConfigForTest.py

```diff
@@ -0,0 +1,26 @@
+from GangaCore.testlib.decorators import add_config
+from GangaCore.Utility.Config import getConfig
+
+# Case 1: Config Section exists & given option also exists -> Outcome expected: Should change the value of the option.
+@add_config([('Configuration', 'user', 'test')])
+def test_add_config_1(gpi):
+    c = getConfig('Configuration')
+    assert c['user'] == 'test'
+
+# Case 2: Config Section exists BUT (.is_open=False) & given option does NOT exists -> Outcome expected: Should not add option to the config.
+@add_config([('Configuration', 'TestOption', 'TestValue')])
+def test_add_config_2(gpi):
+    try:
+        c = getConfig('Configuration')
+        c['TestOption'] == 'TestValue'
+        assert False, "Should throw an Error"
+    except:
+        assert True
```

Comment:
You should test for the specific exception that `c['TestOption']` should raise. I suggest to use [assertRaises](http://docs.python.org/library/unittest.html#unittest.TestCase.assertRaises).

Reply:
I have modified the code for the specific exception, but I am unsure of how to use `assertRaises` here as it needs to inherit `GangaUnitTest`, where as the @add_config decorator needs `gpi` fixture to be used.

Reply:
That is fine as well.